src/Model/Item.php

```diff
@@ -114,6 +114,60 @@ public static function isLegacyMode()
 		return self::$legacy_mode;
 	}
 
+	/**
+	 * Set the pinned state of an item
+	 *
+	 * @param integer $iid    Item ID
+	 * @param integer $uid    User ID
+	 * @param boolean $pinned Pinned state
+	 */
+	public static function setPinned(int $iid, int $uid, bool $pinned)
+	{
+		DBA::update('user-item', ['pinned' => $pinned], ['iid' => $iid, 'uid' => $uid], true);
+	}
+
+	/**
+	 * Get the pinned state
+	 *
+	 * @param integer $iid Item ID
+	 * @param integer $uid User ID
+	 *
+	 * @return boolean pinned state
+	 */
+	public static function getPinned(int $iid, int $uid)
+	{
+		$useritem = DBA::selectFirst('user-item', ['pinned'], ['iid' => $iid, 'uid' => $uid]);
+		if (!DBA::isResult($useritem)) {
+			return false;
+		}
+		return (bool)$useritem['pinned'];
```

Comment:
I think a `?? false` would add more safety in case it's not set. Because it's a nullable field

Reply:
a field that is `null` should be returned as `false` here as well - what the casting does perfectly here.